lib/compilation_engine.rb

```diff
@@ -129,5 +280,14 @@ def output_token
     end
 
     @output.puts("<#{token_type}> #{CGI.escapeHTML(token)} </#{token_type}>")
+    advance
+  end
+
+  def is_symbol_token_and_equal?(symbol)
```

Comment:
Extracting this helper was a great idea. Two nitpicks about its name:

1. It’s too long — it’s fairly obvious what it does so we don’t need to spell out its purpose to this degree.
2. It begins with `is_`, which some style guides [recommend against](https://rubystyle.guide/#bool-methods-prefix) for predicates because it’s redundant and inconsistent with the naming scheme in Ruby’s standard library (e.g. it’s `String#empty?`, not `String#is_empty?`).

So how about something like `#symbol_token?` or even just `#symbol?`?